src/pages/api/deleteCategory.ts

```diff
@@ -0,0 +1,24 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import connectDB from '../../lib/mongodb';
+import { Category } from '@/types/Category';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
```

Comment:
you also need to take action for the merchant in this case. You have a few options:
1. delete all the merchants that are linked to this category.
2. Leave merchans withouth a category and make it visible in the UI and add a functionality to connect merchant to a category.
3. Assign merchants to a specific 'uncategorised' category.
4. Do not allow the user to delete a category that has merchants in it. This will force the user to reassign all the merchants from this category to another before it tries to delete it.